src/HotChocolate/Core/src/Types.Scalars/EmailAddressType.cs

```diff
@@ -0,0 +1,120 @@
+using System;
+using System.Text.RegularExpressions;
+using HotChocolate.Language;
+using HotChocolate.Types.Scalars;
+
+namespace HotChocolate.Types
+{
+    /// <summary>
+    /// The `EmailAddress` scalar type represents a email address, represented as UTF-8 character sequences.
+    /// The scalar follows the specification defined in RFC 5322
+    /// </summary>
+    public class EmailAddressType : StringType
+    {
+        /// <summary>
+        /// Well established regex for email validation
+        /// Source : https://emailregex.com/
+        /// </summary>
+        private static readonly string _validationPattern = @"(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|""(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*"")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])";
+
+        private static readonly Regex _validationRegex =
+            new Regex(
+                _validationPattern,
+                RegexOptions.Compiled | RegexOptions.IgnoreCase);
+        public EmailAddressType()
+            : this(
+                WellKnownScalarTypes.EmailAddress,
+                ScalarResources.EmailAddress_Description)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EmailAddressType"/> class.
+        /// </summary>
+        public EmailAddressType(
+            NameString name,
+            string? description = null,
+            BindingBehavior bind = BindingBehavior.Explicit)
+            : base(name, description, bind)
+        {
+            Description = description;
+        }
+
+        /// <inheritdoc />
+        protected override bool IsInstanceOfType(string runtimeValue)
+        {
+            return runtimeValue != string.Empty && _validationRegex.IsMatch(runtimeValue);
```

Comment:
Does IsMatch behave differentlyu when you pass in a empty string?
Because if not you can justt use `_validationRegex.IsMatch(runtimeValue);` and leave the check on emptystring away :)